testing/events.go

```diff
@@ -140,25 +141,25 @@ func AssertEvents(
 	expected EventsMap,
 	actual sdk.Events,
 ) {
-	hasEvents := make(map[string]bool)
-	for eventType := range expected {
-		hasEvents[eventType] = false
-	}
-
-	for _, event := range actual {
-		expEvent, eventFound := expected[event.Type]
-		if eventFound {
-			hasEvents[event.Type] = true
-			suite.Require().Len(event.Attributes, len(expEvent))
-			for _, attr := range event.Attributes {
-				expValue, found := expEvent[string(attr.Key)]
-				suite.Require().True(found)
-				suite.Require().Equal(expValue, string(attr.Value))
+	for eventType, expEvents := range expected {
+		for _, expEvent := range expEvents {
+			var expEventsOk []bool
+
+			for _, event := range actual {
+				if event.Type == eventType {
+					ok := suite.EqualValues(len(event.Attributes), len(expEvent))
+					for _, attr := range event.Attributes {
+						expValue, found := expEvent[string(attr.Key)]
+						ok = ok && suite.EqualValues(true, found)
+						ok = ok && suite.EqualValues(expValue, string(attr.Value))
```

Comment:
For some reason when I run the test locally it seems like `suite.EqualValues` stops the test if the condition is not satisfied. I just want it to return a boolean... I can do the comparison manually otherwise.